File: python/examples/close_position.py
#!/usr/bin/env python3
"""
Close Position Example

Close an open position completely. The SDK figures out the size and direction.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python close_position.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Close Position Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Close BTC position (if any)
    # The SDK queries your position and builds the counter-order automatically
    try:
        result = sdk.close_position("BTC")
        print(f"Closed position: {result}")
    except Exception as e:
        print(f"No position to close or error: {e}")


if __name__ == "__main__":
    main()

File: typescript/examples/close_position.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Close Position Example
 *
 * Close an open position completely. The SDK figures out the size and direction.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Close BTC position (if any)
  // The SDK queries your position and builds the counter-order automatically
  try {
    const result = await sdk.closePosition("BTC");
    console.log(`Closed position: ${result}`);
  } catch (e) {
    console.log(`No position to close or error: ${e}`);
  }
}

main().catch(console.error);
